docs/pods/storage/model-caching.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "Model Caching"
+description: "Model caching on RunPod allows you to efficiently manage and reuse machine learning models and datasets across multiple pods or serverless instances."
+sidebar_position: 9
+---
+
+
+Model caching on RunPod allows you to efficiently manage and reuse machine learning models and datasets across multiple pods or serverless instances. 
+RunPod significantly reduces download times, network usage, and overall resource consumption when working with large models or datasets from Hugging Face.
+
+## Setting Up Model Caching
+
+To use model caching on RunPod, you need to:
+
+1. Set up the appropriate environment variables.
+2. Create a network volume for the network cache.
+3. Launch your pod or serverless instance with the network volume attached.
+
+## Environment Variables
+
+Configure model caching using the following environment variables:
+
+- `HUGGINGFACE_MODEL` or `HUGGINGFACE_MODELS`: Comma-separated list of models to download (format: `model[:branch]`).
+- `HUGGINGFACE_DATASET` or `HUGGINGFACE_DATASETS`: Comma-separated list of datasets to download (format: `dataset[:branch]`).
+- `HUGGINGFACE_TOKEN`: Token for accessing private models or datasets.
+- `HUGGINGFACE_USER`: Username for accessing private models or datasets.
+
+Examples:
+```
+HUGGINGFACE_MODEL="openchat/openchat-3.5-0106,openchat/openchat-3.5-1210"
+HUGGINGFACE_DATASET="google-bert/bert-base-uncased"
+```
+
+## Using Network Volumes
```

Comment:
all of this has nothing to do with the modelcache, as it behaves out of the users scope. no network volume is needed.

instead we would need to know where the files are actually available after the container is starting:

* modelcache downloads the models / datasets from huggingface
* only when this is done, the pod / worker can actually start and execute what ever their purpose is (like running ComfyUI or doing inference for an LLM)
* the model is mounted read-only in `/runpod/cache/model/<organization>/<repo>/<branch>`
* users can then use the model / dataset inside their project and don't have to bake models into docker images anymore
* public models are shared with all users in the same datacenter
* private models can only be accessed by users who have access to the repo using their `RUNPOD_HUGGINGFACE_USER` + RUNPOD_HUGGINGFACE_TOKEN`. we always check the auth when users try to access this model. the model can then be found at `/runpod/cache/model/<user>/<organization>/<repo>/<branch>`